code/game/objects/structures/window.dm

```diff
@@ -599,3 +600,93 @@
 
 /obj/structure/window/reinforced/clockwork/fulltile/unanchored
 	anchored = FALSE
+
+/obj/structure/window/paperframe
+	name = "paper frame"
+	desc = "A fragile separator made of thin wood and paper."
+	icon = 'icons/obj/smooth_structures/paperframes.dmi'
+	icon_state = "frame"
+	dir = FULLTILE_WINDOW_DIR
+	opacity = TRUE
+	max_integrity = 15
+	obj_integrity = 15
+	fulltile = TRUE
+	flags = PREVENT_CLICK_UNDER
+	smooth = SMOOTH_TRUE
+	canSmoothWith = list(/obj/structure/window/paperframe, /obj/structure/mineral_door/paperframe)
+	glass_amount = 2
+	glass_type = /obj/item/stack/sheet/paperframes
+	heat_resistance = 233
+	decon_speed = 10
+	CanAtmosPass = ATMOS_PASS_YES
+	resistance_flags = FLAMMABLE
+	armor = list(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 0, acid = 0)
+	breaksound = 'sound/items/poster_ripped.ogg'
+	hitsound = 'sound/weapons/slashmiss.ogg'
+	var/static/mutable_appearance/torn
+	var/static/mutable_appearance/paper
+
+/obj/structure/window/paperframe/Initialize()
+	. = ..()
+	torn = mutable_appearance('icons/obj/smooth_structures/paperframes.dmi',icon_state = "torn", layer = ABOVE_OBJ_LAYER - 0.1)
+	paper = mutable_appearance('icons/obj/smooth_structures/paperframes.dmi',icon_state = "paper", layer = ABOVE_OBJ_LAYER - 0.1)
+	for(var/obj/item/I in debris)
+		debris -= I
+		qdel(I)
+
+	var/papers = rand(1,4)
+	debris += new /obj/item/stack/sheet/mineral/wood()
+	for(var/i in 1 to papers)
+		debris += new /obj/item/weapon/paper/natural()
+	update_icon()
+
+/obj/structure/window/paperframe/attack_hand(mob/user)
+	if(!can_be_reached(user))
+		return
+	add_fingerprint(user)
+	if(user.a_intent != INTENT_HARM)
+		user.changeNext_move(CLICK_CD_MELEE)
+		user.visible_message("[user] knocks on [src].")
+		playsound(loc, "pageturn", 50, 1)
+	else
+		take_damage(4,BRUTE,"melee", 0)
+		playsound(loc, hitsound, 50, 1)
+		if(!QDELETED(src))
+			user.visible_message("<span class='danger'>[user] tears a hole in [src].</span>")
+			update_icon()
+
+/obj/structure/window/paperframe/update_icon()
+	if(obj_integrity < max_integrity)
+		cut_overlay(paper)
+		add_overlay(torn)
+		set_opacity(0)
+	else
+		cut_overlay(torn)
+		add_overlay(paper)
+		set_opacity(1)
+	queue_smooth(src)
+
+
+/obj/structure/window/paperframe/attackby(obj/item/weapon/W, mob/user)
+	if(istype(W, /obj/item/weapon/weldingtool))
+		var/obj/item/weapon/weldingtool/WT = W
+		if(WT.isOn())
+			fire_act(W.is_hot())
```

Comment:
I think weldingtool/is_hot returns FALSE if it's off. Remove the weldingtool specific stuff